### DISCUSSION TITLE
All citation keys changed

### DISCUSSION THREAD

**Discussion by andersorder** [retorquere/zotero-better-bibtex#3045] [Q&A]
Recently, it seems like all my Betterbibtex keys changed. I have a deadline for my dissertation on Wednesday and would not like to change about 200 entries manually. Does this have to do with the recent update Zotero or Betterbibtex? Does anyone know how to fix it?

**Comment by retorquere** [comment]
Nothing in BBT or zotero updates changes keys. Only two causes I can think of for mass changes; mass updates to items, or the deletion/corruption of the bbt database.

The best way to guard against this is to autopin keys, but that won't help after the fact.

The database is not on a cloud drive?

**Comment by andersorder** [comment]
Thank you. I just think the database was temporarily corrupted for some reason. It works now. Should think about pinning them for my next project. Does that require betterbibtex?

**Reply by retorquere** [reply]
Yes. https://retorque.re/zotero-better-bibtex/installation/preferences/index.html#automatically-pin-citation-key-after

**Comment by Timver71** [comment]
How long did it take for your citation keys to be right again? All of my citation keys have recently changed as well.

**Reply by retorquere** [reply]
This for an entirely different version of Zotero. A lot has changed since. For one, the keys no longer live in the BBT database, they live in the Zotero database now, so the pinning solution is no longer applicable.

**Reply by Timver71** [reply]
I see. 

I am using Zotero and Better BibTeX in my Overleaf document. All my citation keys in overleaf (when automatically imported) used to look something like: schmidl_anomaly_2022

Now they look like: schmidlAnomalyDetectionTime2022

Do you have any idea if I could revert the citation key format to what it was before it changed? Any help is much appreciated. 

Thanks,
Tim

**Reply by retorquere** [reply]
Ah. Yes. Set your citation key formula to `zotero`, select all items, right-click, BBT, regen.

**Reply by retorquere** [reply]
The keys changing is a direct result of the keys moving to the new Zotero field.

**Reply by Timver71** [reply]
After a few refreshes that fixed everything. Thank you so much! 

I want to ensure that my citation keys dont undergo a mass format change that I cant fix , though I dont mind just changing format back to 'zotero' now that I know about it. Do I need to pin all citation keys (does pining still work, you seemed to say pining is no longer a solution?) or do you think the 'zotero' format will stay consistent? I am happy to ask on the Zotero forums if you are unsure!

thanks for your help :)

**Reply by retorquere** [reply]
> I want to ensure that my citation keys dont undergo a mass format change that I cant fix , though I dont mind just changing format back to 'zotero' now that I know about it. Do I need to pin all citation keys (does pining still work, you seemed to say pining is no longer a solution?)

It is no longer a solution to the problem @andersorder was experiencing. The problem arose from his BBT database being corrupted -- pinning puts a copy of the key in the zotero database (in the `extra` field). Now that zotero has proper citation key fields, the keys always live in the zotero database.

Pinning was something that was really for the old situation -- citation keys lived in a local database that was not shared in sync, and pinning put the citation keys in the extra field so they'd sync. That is now no longer the case, the old pinning behavior is sort of the default behavior now, as they just live in their native field. There are some vestiges of the old behavior present but I wouldn't really recommend using it.

> or do you think the 'zotero' format will stay consistent?

Pretty sure, but the existing keys are entirely stable now, regardless of the formula in use, as they are saved in the item.

**Reply by Timver71** [reply]
Ok, thanks again for all the help. I really appreciate it!
